extensions/wikidata/src/org/openrefine/wikidata/utils/EntityCache.java

```diff
@@ -39,24 +39,23 @@
 
 public class EntityCache {
 
-    private static EntityCache _entityCache = new EntityCache(BasicApiConnection.getWikidataApiConnection());
+    private static Map<String, EntityCache> entityCacheMap = new HashMap<>();
 
-    private LoadingCache<String, EntityDocument> _cache = null;
-    private WikibaseDataFetcher _fetcher;
+    private LoadingCache<String, EntityDocument> cache;
+    private WikibaseDataFetcher fetcher;
 
-    protected EntityCache(ApiConnection connection) {
-        this(new WikibaseDataFetcher(connection, Datamodel.SITE_WIKIDATA));
+    protected EntityCache(String entityPrefix, String mediaWikiApiEndpoint) {
+        this(new WikibaseDataFetcher(new BasicApiConnection(mediaWikiApiEndpoint), entityPrefix));
     }
     
     protected EntityCache(WikibaseDataFetcher fetcher) {
-        _fetcher = fetcher;
+        this.fetcher = fetcher;
 
-        _cache = CacheBuilder.newBuilder().maximumSize(4096).expireAfterWrite(1, TimeUnit.HOURS)
+        cache = CacheBuilder.newBuilder().maximumSize(4096).expireAfterWrite(1, TimeUnit.HOURS)
                 .build(new CacheLoader<String, EntityDocument>() {
 
-                    public EntityDocument load(String entityId)
-                            throws Exception {
-                        EntityDocument doc = _fetcher.getEntityDocument(entityId);
+                    public EntityDocument load(String entityId) throws Exception {
```

Comment:
This is a very wide `throws` statement. Can it be narrowed at all to a smaller subset of `Exception`s?